packages/typescript-estree/src/parser.ts

```diff
@@ -287,19 +290,25 @@ function applyParserOptionsToExtra(options: TSESTreeOptions): void {
     extra.projects = options.project;
   }
 
+  if (typeof options.tsconfigRootDir === 'string') {
```

Comment:
there was a bug: glob project paths were resolved relative to CWD, not `tsconfigRootDir`.